### PR TITLE
Added EXT-X-CUE-OUT and EXT-X-CUE-IN media segment tags

### PR DESCRIPTION
**Pull request by JohanSpaedtke**
Title says it all. Added the MediaSegmentTags
#EXT-X-CUE-OUT
#EXT-X-CUE-IN

### CODE REVIEW

**Comment by carlanton** [comment]
Cool stuff! However, I believe that the correct type of cueOut() should be `Optional<Double>`?
Example: https://docs.aws.amazon.com/medialive/latest/ug/sample-manifests-hls.html

Also: make sure that your changes are formatted accordingly :)

**Comment by JohanSpaedtke** [comment]
The more you know... The documentation I'm working from has it as an int but some googling also shows the following
#EXT-X-CUE-OUT:DURATION=30
Any idea how to support all of them in a sane way? 

Regarding formatting. Sure what did I do wrong :) ?

**Comment by carlanton** [comment]
Ok, there seems to be multiple styles for ad markers. Let's use the format you proposed (Elemental style?). If someone need to support more styles, we can figure it out later :)

File: src/main/java/io/lindstrom/m3u8/model/MediaSegment.java [hunk 1]
```diff
@@ -30,6 +30,13 @@
     default boolean discontinuity() {
         return false;
     }
+    Optional<Integer> cueOut();
```
Comment (carlanton) [inline src/main/java/io/lindstrom/m3u8/model/MediaSegment.java:33]: insert newline

File: src/main/java/io/lindstrom/m3u8/model/MediaSegment.java [hunk 2]
```diff
@@ -30,6 +30,13 @@
...
+
+    @Value.Default
+    default boolean cueIn()
+    {
```
Comment (carlanton) [inline src/main/java/io/lindstrom/m3u8/model/MediaSegment.java:37]: `{` should be on previous line

**Comment by JohanSpaedtke** [comment]
Sure :) 
You had some issues with the formatting? What do you want med to fix?

File: src/main/java/io/lindstrom/m3u8/parser/MediaSegmentTag.java [hunk 3]
```diff
@@ -60,6 +60,32 @@ public void write(MediaSegment mediaSegment, TextBuilder textBuilder) {
         }
     },
 
+    EXT_X_CUE_OUT {
+        @Override
+        public void read(MediaSegment.Builder builder, String attributes){
```
Comment (carlanton) [inline src/main/java/io/lindstrom/m3u8/parser/MediaSegmentTag.java:65]: `){` should be `} {`

**Review by carlanton — CHANGES_REQUESTED** [review]

**Review by carlanton — APPROVED** [review]

**Comment by carlanton** [comment]
Your change is included in version 0.15. It should be on maven soon :)

**Pull request merged**